email/Email.php

```diff
@@ -389,15 +496,15 @@ protected function parseVariables($isPlain = false) {
 	}
 
 	/**
-	 * Validates the email address. Returns true of false
+	 * Validates the email address according to RFC 822 standard.
+	 *
+	 * See: http://php.net/manual/en/filter.filters.validate.php
+	 *
+	 * @param string $address
+	 * @return bool
 	 */
 	public static function validEmailAddress($address) {
-		if (function_exists('filter_var')) {
-			return filter_var($address, FILTER_VALIDATE_EMAIL);
-		} else {
-			return preg_match('#^([a-zA-Z0-9_+\.\-]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)'
-				. '|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$#', $address);
-		}
+		return filter_var($address, FILTER_VALIDATE_EMAIL) !== false;
```

Comment:
I'm slightly concerned that this counts as an API change more than a bug fix - do we know how this will effect valid/invalid email addresses getting through?